EIPS/eip-draft_permission_token.md

```diff
@@ -0,0 +1,317 @@
+---
+eip: draft_permission_token
+title: Permission Token
+description: A new token that held the permission of an address in an ecosystem
+author: Chiro (@chiro-hiro)
+discussions-to: https://ethereum-magicians.org/t/eip-644-a-standard-for-permission-token/9105
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-01-19
+---
+
+## Abstract
+
+A new token standard that held the permission of an address in an ecosystem. A permission token can be transferred/delegated by the permission owner to an grantee's address.
+
+## Motivation
+
+You may still encounter special roles like `Owner`, `Operator`, `Manager`, `Validator` in many existing smart contracts because we're still using permissioned addresses to operate and manage them. It is difficult to audit and maintain since these permissions are not managed centrally in a single smart contract.
+
+To secure the communication between smart contracts, this EIP provides a straightforward, centralized, and scalable method for managing permissions.
+
+## Specification
+
+We RECOMMENDED to define permissions are power of 2, that allowed us to create up to `2^256` different roles.
```

Comment:
Why not use `bytes32` and use keccak hashes of strings? Also, wouldn't this limit the specification to 256 roles?

Reply:
[EIP-5982](https://eips.ethereum.org/EIPS/eip-5982) is based on roles meanwhile EIP-6366 is based on permissions (bottom up approach).  

Apparently, `keccak256(string)` is cost more than a bitwise operator on `uint256` that's why I supposed to use `uint256` to store permissions of an account. We can define a single permission is power of 2 (1 bit is represented for 1 permission) and then combine them with `OR` operator to create roles. 

**Example 1**, Here is how we define permissions and roles.

```
const PERMISSION_NONE = 0;
const PERMISSION_READ = 1; // 2⁰
const PERMISSION_WRITE = 2; // 2¹
const PERMISSION_EXECUTE = 4; // 2²

// Role admin = 4 | 2 | 1 = 7
const ROLE_ADMIN = PERMISSION_READ | PERMISSION_WRITE | PERMISSION_EXECUTE;
// Role operator = 1 | 2 = 3
const ROLE_OPERATOR = PERMISSION_READ | PERMISSION_WRITE;
```

**Example 2**, The verification is much more flexible and efficient, you can check a single permission or a subset of permission by using `AND` operator.

```
checkingPermission & ROLE_OPERATOR == ROLE_OPERATOR
```

A single permission token can define 256 permissions and 2²⁵⁶ roles.

Reply:
Huh. That's neat!